pkg/store/prometheus.go

```diff
@@ -496,9 +500,51 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 		return &storepb.LabelValuesResponse{Values: []string{l}}, nil
 	}
 
-	vals, err := p.client.LabelValuesInGRPC(ctx, p.base, r.Label, nil, r.Start, r.End)
-	if err != nil {
-		return nil, err
+	var (
+		sers []map[string]string
+		err  error
+	)
+
+	lvc := false // LabelValuesCall
+	vals := []string{}
+	v := p.promVersion()
+
+	version, err := semver.Parse(v)
+	if err == nil {
+		baseVer, err := semver.Make("2.24.0")
+		if err != nil {
+			return nil, err
+		}
+		if version.Compare(baseVer) == 1 {
+			lvc = true
+		}
+	}
+
+	if len(r.Matchers) == 0 || lvc {
```

Comment:
```suggestion
	// Make sure the prometheus version is a valid semver type.
	version, err := semver.Parse(v)
	if err != nil {
	   return nil, err
	}
	
        baseVer, err := semver.Make("2.24.0")
        if err != nil {
	   return nil, err
        }
	
	if len(r.Matchers) == 0 || version.Compare(baseVer) == 1 {
```

Reply:
Lets simplify things, return early if an error - https://thanos.io/tip/contributing/coding-style-guide.md/#control-structure-prefer-early-returns-and-avoid-else

Reply:
I think if the version is malformed we can still make a series call instead of returning the error. Please see `If it is malformed, we just use the series way` in https://github.com/thanos-io/thanos/pull/3940#discussion_r621221476